LinkedList/LinkedList/TSLList.h

```diff
@@ -0,0 +1,350 @@
+#pragma once
+#include <functional>
+
+template <typename Type>
+class TSLList
+{
+private:
+	int Size;
+	struct FNode
+	{
+		Type Element;
+		FNode* Next = nullptr;
+	};
+	FNode* Head = nullptr;
+	FNode* Tail = nullptr;
+
+public:
+	TSLList()
+	{
+		Size = 0;
+	}
+
+	~TSLList()
+	{
+		Clear();
+	}
+
+	//-------------------------------CopyConstructor
+	TSLList(const TSLList<Type>& OtherTSLlist)
+	{
+		FNode* Temporal = OtherTSLlist.Head;
+
+		for (int i = 0; i < Size; i++)
```

Comment:
Ok, so discussing it with Santiago, there are actually two problems here. First, since you're not initialising Size on declaration, as it is a built-in type it will become whatever memory used to occupy its place. So, for all we know Size might be 8'000.000, so you're potentially trying to do something eight million times and this might also call trouble since you're likely soon to set your temporal variable to nullptr, so it should give you that error too.

Reply:
The second problem has to do with the fact that you want to iterate untill i reaches OtherTSLlist.Size, not the current Size, which should be 0.